crypto_hasher.go

```diff
@@ -108,6 +108,10 @@ func (ch *cryptoHasher) Write(p []byte) (n int, err error) {
 	return ch.hashAlgorithm.Write(p)
 }
 
+func (ch *cryptoHasher) Update(bytes []byte) {
```

Comment:
Please remove this and rename 

`func (ch *cryptoHasher) Write(p []byte) (n int, err error)`

to:

`func (ch *cryptoHasher) Update(p []byte) (n int, err error) {`

Reply:
Upon further investigation, it seems more idiomatic to have a function `Write` to update the hash instead of `Update`. Lets rename all instances of `Update` to `Write`

Reply:
Renamed to Write().